controllers/networkplugins_controller.go

```diff
@@ -481,12 +486,10 @@ func (ovsConfig *OvsT) WriteConfigToTemplate(outputDir, registry string) error {
 	}
 
 	if ovsConfig.DPDK != nil {
-                if ovsConfig.DPDK.LcoreMask == "" || ovsConfig.DPDK.SocketMem == "" || ovsConfig.DPDK.PmdCpuMask == "" || ovsConfig.DPDK.HugepageMemory == "" || ovsConfig.DPDK.HugepageSize == "" {
-                        return fmt.Errorf("LcoreMask, SocketMem, PmdCpuMask, HugepageMemory and HugepageSize are required parameters to enable Dpdk")
+                if ovsConfig.DPDK.LcoreMask == "" || ovsConfig.DPDK.SocketMem == "" || ovsConfig.DPDK.PmdCpuMask == "" || ovsConfig.DPDK.HugepageMemory == "" {
+                        return fmt.Errorf("LcoreMask, SocketMem, PmdCpuMask, HugepageMemory are required parameters to enable Dpdk")
                 }
-		if ovsConfig.DPDK.HugepageSize != "2Mi" && ovsConfig.DPDK.HugepageSize != "1Gi" {
-			return fmt.Errorf("HugepageSize must be one of the allowed huge page sizes: 2Mi and 1Gi")
-		}
+		config["HugepageSize"] = GetHugepageSize()
```

Comment:
lets keep it as part of ovsConfig.DPDK.HugepageSize since this is in DPDK context anyway?

Reply:
For that (ovsConfig.DPDK.HugepageSize), it needs to be part of CRD.